kolibri/plugins/facility/assets/src/modules/userManagement/handlers.js

```diff
@@ -1,22 +1,99 @@
 import pickBy from 'lodash/pickBy';
-import FacilityUserResource from 'kolibri-common/apiResources/FacilityUserResource';
 import samePageCheckGenerator from 'kolibri-common/utils/samePageCheckGenerator';
+import FacilityUserResource from 'kolibri-common/apiResources/FacilityUserResource';
 import { _userState } from '../mappers';
+
+export function fetchSortedFacilityUsersHandler(
+  store,
+  { column, order, page, page_size, router, search, user_type },
+) {
+  store.commit('SET_STATE', { dataLoading: true });
+
+  const orderingParam = order === 'desc' ? `-${column}` : column || null;
+  const shouldResolve = samePageCheckGenerator(store);
+
+  return FacilityUserResource.fetchCollection({
+    getParams: pickBy({
+      ordering: orderingParam,
+      page,
+      page_size,
+      search,
+      user_type,
+    }),
+    force: true,
+  })
+    .then(users => {
+      if (shouldResolve()) {
+        const mappedUsers = users.results.map(_userState);
+        store.commit('SET_STATE', {
+          facilityUsers: mappedUsers,
+          totalPages: users.total_pages,
+          usersCount: users.count,
+        });
+
+        const currentQuery = router.currentRoute.query;
+        const newQuery = pickBy({
+          ...currentQuery,
+          ordering: column,
+          order,
+          page,
+          page_size,
+          search,
+          user_type,
+        });
+
+        router.push({
+          path: router.currentRoute.path,
+          query: newQuery,
+        });
+      }
+      store.commit('SET_STATE', { dataLoading: false });
+      store.dispatch('userManagement/notLoading');
```

Comment:
I see console errors related to this because `notLoading` is a core/root module action so you can just do `store.dispatch('notLoading') throughout.

I see the same errors for other action dispatches:

```
index.vue:383 [vuex] unknown action type: userManagement/showUserPage
handlers.js:55 [vuex] unknown local action type: handleApiError, global type: userManagement/handleApiError
handlers.js:58 [vuex] unknown local action type: userManagement/notLoading, global type: userManagement/userManagement/notLoading
```

Reply:
I did try both removing/adding userManagement with these action dispatches, but the error just doesn't resolve. Also, added `{root: true}` at the end - but that crashed the website somehow. If you could let me know of ways about how this can be fixed, it would be really helpful. Thanks!
![image](https://github.com/user-attachments/assets/4edad6f7-7def-423d-947f-c05746d97cb6)

Reply:
Thanks all.

@BabyElias unless you figure it out by then, we can have a look at this together tomorrow during our check-in :)

Reply:
@nucleogenesis @LianaHarris360 
The `unknown action type showUserPage` error has been resolved. However, I tried multiple strategies for the other errors, but to no avail. If this is not a blocking issue for this PR (since all functionalities are intact), maybe this one can be merged and then the errors can be dealt with in subsequent PRs - perhaps by me, if some additional direction is provided regarding the same.